cli/slsa-verifier/main.go

```diff
@@ -88,39 +104,53 @@ func isFlagPassed(name string) bool {
 	return found
 }
 
-func runVerify(artifactPath, provenancePath, source string,
+func runVerify(artifactImage, artifactPath, provenancePath, source string,
 	branch, builderID, ptag, pversiontag *string,
 ) ([]byte, string, error) {
-	f, err := os.Open(artifactPath)
-	if err != nil {
-		return nil, "", err
-	}
-	defer f.Close()
+	ctx := context.Background()
 
-	provenance, err := os.ReadFile(provenancePath)
+	// Artifact hash retrieval depends on the artifact type.
+	artifactHash, err := getArtifactHash(artifactImage, artifactPath)
 	if err != nil {
 		return nil, "", err
 	}
 
-	h := sha256.New()
-	if _, err := io.Copy(h, f); err != nil {
-		return nil, "", err
-	}
-	artifactHash := hex.EncodeToString(h.Sum(nil))
-
 	provenanceOpts := &options.ProvenanceOpts{
 		ExpectedSourceURI:    source,
 		ExpectedBranch:       branch,
-		ExpectedDigest:       artifactHash,
 		ExpectedVersionedTag: pversiontag,
+		ExpectedDigest:       artifactHash,
 		ExpectedTag:          ptag,
 	}
 
 	builderOpts := &options.BuilderOpts{
 		ExpectedID: builderID,
 	}
 
-	ctx := context.Background()
-	return verifiers.Verify(ctx, provenance,
-		artifactHash, provenanceOpts, builderOpts)
+	var provenance []byte
+	if provenancePath != "" {
+		provenance, err = os.ReadFile(provenancePath)
+		if err != nil {
+			return nil, "", err
+		}
+	}
+
+	return verifiers.Verify(ctx, artifactImage, provenance, artifactHash, provenanceOpts, builderOpts)
```

Comment:
do we want to have a `VerifyArtifact()` and `VerifyImage()`?

Reply:
Can do in the sub-command splitting issue!

Reply:
Noted in the issue